internal/agent/hooks/logs.go

```diff
@@ -17,9 +21,31 @@ import (
 type CopyLogs struct{}
 
 func (k CopyLogs) Run(c config.Config, _ v1.Spec) error {
+	c.Logger.Logger.Debug().Msg("Running CopyLogs hook")
 	c.Logger.Debugf("Copying logs to persistent partition")
 	_ = machine.Umount(constants.PersistentDir)
 
+	// Path if we have encrypted persistent
+	if len(c.Install.Encrypt) != 0 {
+		err := kcrypt.UnlockAll(false)
```

Comment:
We are in `CopyLogs` and we are decrypting partitions? This feels wrong...

Reply:
yes, if you have encrypted partitions you need to decrypt them to be able to work on them, otherwise you cant copy the logs on them.

If our mount would auto do this, it would be nice :D

Reply:
gotcha